internal/proxy/proxy.go

```diff
@@ -195,8 +198,17 @@ func (p *Proxy) handleJenkinsUIRequest(w http.ResponseWriter, r *http.Request) (
 					err = p.processTemplate(w, ns)
 					p.RecordStatistics(pci.NS, time.Now().Unix(), 0) //FIXME - maybe do this at the beginning?
 				} else { //If Jenkins is running, remove the cookie
-					cookie.Expires = time.Unix(0, 0)
-					http.SetCookie(w, cookie)
+					var resp *http.Response
+					//OpenShift can take up to couple tens of second to update HAProxy configuration for new route
+					//so even if the pod is up, route might still return 500 - i.e. we need to check the route
+					//before claiming Jenkins is up
+					resp, err = p.loginJenkins(pci, "")
+					if resp.StatusCode == 200 || resp.StatusCode == 403 {
```

Comment:
This might panic. You are effectively ignoring the error here. If there is an underlying error, the response will be nil in which case the call to _resp.StatusCode_ will panic. Usually in Go if a method returns an error, you always have to first check the error before doing anything else with any other returned object. More often than not, if an error occurs other return parameters are nil.

Reply:
Yeah, sorry, I over optimized there - leaving the error check for later, but obviously resp == nil would cause panic, thanks